Update zenburn theme

jakobj opened the pull request:
As the title suggests.

Building on @theli-ua 's work, I've attempted to faithfully port the complete color palette and map them to tree-sitter scopes. I like the way it turned out, but have only checked python, rust, and markdown highlighting, so this PR could benefit from additional set of eyes and maybe some testing, e.g., by @theli-ua :)

ps: also included the missing license file

jakobj commented:
Thanks for the timely feedback @theli-ua ! I've just updated the PR with all your suggestions. Here's what it looks like now
<img width="1230" height="1345" alt="2025-09-16-223123_1230x1345_scrot" src="https://github.com/user-attachments/assets/ee55b276-2943-48e4-b8d7-9bb90543afc6" />

Re: license I've just followed the guidlines here https://github.com/helix-editor/helix/tree/master/runtime/themes#user-submitted-themes

Might also be appropriate to include you explicitly as an author in the theme file, what do you think?

jakobj commented:
- Imports: the new version seems to be as close to vim as I can make it (using the `Include` colors for the tree-sitter `namespace` node). Any suggestions on how to better match the original theme are appreciated :pray: 
- Linenr: Good catch! Since we're trying to create a faithful port, I've updated this to match the original theme. This also forced me to change the `gutter` background, hope this doesn't break anything else. Looks like this now
<img width="433" height="261" alt="2025-09-18-003800_433x261_scrot" src="https://github.com/user-attachments/assets/73e20734-55da-483e-89ad-38f41ba04dd9" />

- True/False: I think these are not bold in the original theme, e.g. from gvim
<img width="368" height="223" alt="2025-09-18-003527_368x223_scrot" src="https://github.com/user-attachments/assets/61f9b718-b160-4e37-b10d-26f4ba0d1809" />

as these are styled by the `Boolean` color, see https://github.com/jnurmine/Zenburn/blob/master/colors/zenburn.vim#L244 and https://vimhelp.org/syntax.txt.html#%3Asyn-files

- Statusline: Good catch, fixed! Apparently in vim the statusline colors are [reversed](https://stackoverflow.com/a/36430577), because why not ;)

theli-ua commented:
Latest commit seem to have changed colors for git changed/edited/added lines
master:
<img width="84" height="247" alt="image" src="https://github.com/user-attachments/assets/2e7e333b-b4a8-4610-ac71-ea1b4724c2f9" />


pr:
<img width="91" height="380" alt="image" src="https://github.com/user-attachments/assets/e0381b39-8c61-4fdd-9b13-3766eeaf8266" />


I this intentional?


Also, it's been few years sine I've used vim/zenburn, is that warning/hing bright white color how it was in the original theme?

<img width="669" height="74" alt="image" src="https://github.com/user-attachments/assets/1675614d-db7b-441c-a5e4-c3a77448d151" />

jakobj commented:
- git: This was not intentional, fixed now. Needed to style highlighting in diff view and gutter separately. I copied your suggestion to use red (string) color to highlight `diff.minus` in the gutter.
- warning/hint: Not sure since I moved from emacs to helix and emacs zenburn is quite different from the original theme. I just used the `Todo` color [from the original theme](https://github.com/jnurmine/Zenburn/blob/master/colors/zenburn.vim#L297C4-L297C8). We could use something darker if you prefer?
- statusline: Apparently the "reversed" modifier from the statusline also affect the color for the error bullet. Not sure this is intentional(?), but I've worked around it now by manually reversing fg/bg of the status line.
